### PR TITLE
Flaky/deactivate async code

### PR DESCRIPTION
**Pull request by scherler**
# Description

Try to reduce the flaky tests
@reviewbybees

### CODE REVIEW

**Comment by scherler** [comment]
https://ci.blueocean.io/job/ATH-Jenkinsfile/job/master/662/console
and https://ci.blueocean.io/job/ATH-Jenkinsfile/job/master/663/console are provoked without this change

**Comment by michaelneale** [comment]
This seems to have failures to do with test tab: 

```
2016-12-19 23:41:02,575 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner -  [1;31mTEST FAILURE:[0m [0;31m1[0m error during execution,  [0;31m0[0m assertions failed, [0;32m927[0m passed. (7m 29s)
2016-12-19 23:41:02,575 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - 
2016-12-19 23:41:02,575 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - [0;31m ??? multibranch/testResults[0m
2016-12-19 23:41:02,576 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - 
2016-12-19 23:41:02,576 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner -    - Check that the tests tab displays correctly [1;33m(8ms)[0m
2016-12-19 23:41:02,576 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - 
2016-12-19 23:41:02,576 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - [0;31m     Error while running [Multibranch / Test Results / Check that the tests tab displays correctly]:[0m
2016-12-19 23:41:02,576 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - [0;90m   
2016-12-19 23:41:02,576 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner -      Error: resultItemHead"] was not found in "bluePipelineRunDetail". Available elements: code,progressBar,logHeader,fullLog,followAlongOn,followAlongOff,emptystate,emptystateQueued,detailTitle,closeButton,activityTable,highlightedGraph,logConsole,artifactTable,changes,tests,authors,circleSuccess,runningResult,firstResult,firstErrorResult,resultItemHead,testConsole
2016-12-19 23:41:02,576 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner -        at Object.module.exports.commands.expandTest (/home/ubuntu/slave/workspace/e_flaky_deactivateAsyncCode-C5U7E2NMS3KR5TRIIXHIWJ7WVABZHV3VTEPAZGUWMFIFVAWSMROA/src/main/js/page_objects/blueocean/bluePipelineRunDetail.js:443:10)
2016-12-19 23:41:02,576 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner -        at Object.module.exports.Check that the tests tab displays correctly (/home/ubuntu/slave/workspace/e_flaky_deactivateAsyncCode-C5U7E2NMS3KR5TRIIXHIWJ7WVABZHV3VTEPAZGUWMFIFVAWSMROA/src/test/js/multibranch/testResults.js:43:28)[0m
2016-12-19 23:41:02,576 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - 
2016-12-19 23:41:02,576 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - 
2016-12-19 23:41:02,602 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - 
2016-12-19 23:41:02,609 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - [23:41:02] 'default' errored after 7.48 min
2016-12-19 23:41:02,609 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - [23:41:02] Error: Command `nightwatch --suiteRetries 2` failed with exit code 1
2016-12-19 23:41:02,609 [Thread-2] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner -     at ChildProcess.exithandler (child_process.js:206:12)
```

So not sure if it is quite right yet: 

https://ci.blueocean.io/job/ATH-Jenkinsfile/job/flaky%252FdeactivateAsyncCode/2/console

**Comment by scherler** [comment]
@michaelneale hmm maybe it is the failing on headless again. Need to look into that.

**Comment by tfennelly** [comment]
What's the story with these changes? I see commits from all over the place (including from myself).

**Comment by scherler** [comment]
@tfennelly that is because it comes from my fork. They do not appear in the changed files though

File: src/main/js/page_objects/blueocean/bluePipelineRunDetail.js [hunk 1]
```diff
@@ -417,17 +425,25 @@ module.exports.commands = [{
...
     self.waitForElementVisible('.replay-button');
     self.click('.replay-button');
     return self;
+  },
+ /**
+  * Expand the test.
+  */
+  expandTest: function () {
+    const self = this;
+    self.click('@resultItemHead"]', function (result) {
```
Comment (cliffmeyers) [inline src/main/js/page_objects/blueocean/bluePipelineRunDetail.js]: I think this should include a `waitForElementVisible` on `@resultItemHead`. I saw flakiness and frequent test failures when going from a click on the Tests tab immediately to clicking on `@resultItemHead`

File: src/test/js/multibranch/commitMessages.js [hunk 2]
```diff
@@ -15,69 +15,69 @@ const git = require("../../../main/js/api/git");
...
+    //             browser.useXpath().waitForElementVisible(`//tr[@id="${jobName}-2"]/*/a[text()="somefile created"]`);
+    //
+    //             done();
+    //         });
+    //
+    //     }).end();
+    // },
 
-}
+};
```
Comment (cliffmeyers) [inline src/test/js/multibranch/commitMessages.js]: This entire file is commented out...?

File: src/main/js/page_objects/blueocean/bluePipelineRunDetail.js (hunk 1 above)
Comment (scherler) [inline src/main/js/page_objects/blueocean/bluePipelineRunDetail.js]: @cliffmeyers yeah correct. You should always for the element before.

**Comment by tfennelly** [comment]
@scherler what's the status of this PR Thor?

**Comment by michaelneale** [comment]
closing this as old

**Pull request closed without merging**